src/Core/Resolvers/Factories/QueryManagerFactory.cs

```diff
@@ -0,0 +1,109 @@
+        private readonly IDictionary<DatabaseType, IQueryExecutor> _queryExecutors;
+        private readonly IDictionary<DatabaseType, DbExceptionParser> _dbExceptionsParsers;
+
+        /// <summary>
+        /// Initiates an instance of QueryManagerFactory
+        /// </summary>
+        /// <param name="runtimeConfigProvider">runtimeconfigprovider.</param>
+        /// <param name="logger">logger.</param>
+        /// <param name="contextAccessor">httpcontextaccessor.</param>
+        public QueryManagerFactory(RuntimeConfigProvider runtimeConfigProvider, ILogger<IQueryExecutor> logger, IHttpContextAccessor contextAccessor)
+        {
+            _queryBuilders = new Dictionary<DatabaseType, IQueryBuilder>();
+            _queryExecutors = new Dictionary<DatabaseType, IQueryExecutor>();
+            _dbExceptionsParsers = new Dictionary<DatabaseType, DbExceptionParser>();
+            foreach (DataSource dataSource in runtimeConfigProvider.GetConfig().ListAllDataSources())
+            {
+                IQueryBuilder? queryBuilder = null;
+                IQueryExecutor? queryExecutor = null;
+                DbExceptionParser? exceptionParser = null;
+
+                if (_queryBuilders.ContainsKey(dataSource.DatabaseType))
+                {
+                    // we have already created the builder, parser and executor for this database type.No need to create again.
+                    continue;
+                }
+
+                switch (dataSource.DatabaseType)
+                {
+                    case DatabaseType.CosmosDB_NoSQL:
+                        break;
+                    case DatabaseType.MSSQL:
+                        queryBuilder = new MsSqlQueryBuilder();
+                        exceptionParser = new MsSqlDbExceptionParser(runtimeConfigProvider);
+                        queryExecutor = new MsSqlQueryExecutor(runtimeConfigProvider, exceptionParser, logger, contextAccessor);
+                        break;
+                    case DatabaseType.MySQL:
+                        queryBuilder = new MySqlQueryBuilder();
+                        exceptionParser = new MySqlDbExceptionParser(runtimeConfigProvider);
+                        queryExecutor = new MySqlQueryExecutor(runtimeConfigProvider, exceptionParser, logger, contextAccessor);
+                        break;
+                    case DatabaseType.PostgreSQL:
+                        queryBuilder = new PostgresQueryBuilder();
+                        exceptionParser = new PostgreSqlDbExceptionParser(runtimeConfigProvider);
+                        queryExecutor = new PostgreSqlQueryExecutor(runtimeConfigProvider, exceptionParser, logger, contextAccessor);
+                        break;
+                    default:
+                        throw new NotSupportedException(dataSource.DatabaseTypeNotSupportedMessage);
+                }
+
+                _queryBuilders.TryAdd(dataSource.DatabaseType, queryBuilder!);
```

Comment:
TryAdd?

Reply:
Was using tryadd in general as a FailSafe. This was the overall guidance in the previous pr 
<img width="528" alt="image" src="https://github.com/Azure/data-api-builder/assets/124841904/31de4e0e-8b0f-4721-99b5-9a7b6c77e096">
please see here: https://github.com/Azure/data-api-builder/pull/1639/files/3a855c170836e9c0cf21715a7a799808b401e222